frontend/ui/widget/screensaverwidget.lua

```diff
@@ -163,6 +165,9 @@ ScreenSaverWidget.onAnyKeyPressed = ScreenSaverWidget.onClose
 ScreenSaverWidget.onExitScreensaver = ScreenSaverWidget.onClose
 
 function ScreenSaverWidget:onCloseWidget()
+    Device.screen_saver_mode = false
+    Device.screen_saver_lock = false
```

Comment:
Also a bit hard to grasp. They are plugged into Device (as some random holder of these flag), but not really documented, and their use is a bit scattered. Some single place documentation of these 2 flags could help.

Reply:
They've indeed historically been scattered all over the place (but originated in Device) ;o).

Now, at least they're only *set* inside ScreenSaver/ScreenSaverWidget.

It kinda makes sense for them to be in Device, as it *is* a global state flag.

Il'l shove some comments about it... somewhere ;o).